timeout_decorator/timeout_decorator.py

```diff
@@ -135,17 +134,15 @@ def cancel(self):
         """Terminate any possible execution of the embedded function."""
         if self.__process.is_alive():
             self.__process.terminate()
-            raise TimeoutError()
+        raise TimeoutError()
 
     @property
     def ready(self):
         """Read-only property indicating status of "value" property."""
-        if self.__queue.full():
-            return True
-        elif not self.__queue.empty():
-            return True
-        elif self.__timeout < time.time():
+        if self.__timeout < time.time():
             self.cancel()
+        if self.__queue.full() and not self.__queue.empty():
```

Comment:
I think this can be just `return self.__queue.full() and not self.__queue.empty()`